### PR TITLE
Do not load `ScopedValues` with `using`

### PR DESCRIPTION
**Pull request by LilithHafner**
Stop loading `ScopedValues` with `using` so folks use `ScopedValues.with` or `using ScopedValues` rather than `Base.with`.

Implements https://github.com/JuliaLang/julia/pull/55095#issuecomment-2272334437

~Have to bump the StyledStrings stdlib to include https://github.com/JuliaLang/StyledStrings.jl/pull/80~ Done

### CODE REVIEW

File: base/Base.jl [hunk 1]
```diff
@@ -424,7 +424,7 @@ include("weakkeydict.jl")
 
 # ScopedValues
 include("scopedvalues.jl")
-using .ScopedValues
+import .ScopedValues
```
Comment (vchuravy) [inline base/Base.jl:427]: This is not needed then.

**Comment by vchuravy** [comment]
Needs a judgement call from other people if this change is okay so late in the release process for 1.11.
IMO we should minimize breaking changes once a release candidate has been made.

File: base/Base.jl (hunk 1 above)
Comment (LilithHafner) [reply base/Base.jl:427]:
```suggestion 
```

**Comment by LilithHafner** [comment]
Inclusion in a release candidate does not commit us to an API. See @JeffBezanson's comment here: https://github.com/JuliaLang/julia/issues/55249#issuecomment-2272184464.

Being late in a release process should dissuade us from including a significant complexity introduction that may carry bugs or unintended side effects but should not stop us form adjusting API.

Removing the breaking label because this does not break released behavior.

**Comment by nsajko** [comment]
To add to what @LilithHafner says, IMO this can't be breaking anyway because `Base` never declared `with` as `public`:

```julia-repl
help?> Base.with
  │ Warning
  │
  │  The following bindings may be internal; they may change or be removed in future versions:
  │
  │    •  Base.with

  with(f, (var::ScopedValue{T} => val)...)
```

**Comment by KristofferC** [comment]
Still would have been good to do a PkgEval here and do some proactive fixes in the ecosystem instead of letting the next 1.11 PkgEval run fall into despair.

**Pull request merged**